content/300-guides/050-database/100-developing-with-prisma-migrate/180-generating-down-migrations.mdx

```diff
@@ -0,0 +1,146 @@
+
+This section describes how to generate a down migration SQL file along with the corresponding up migration, and then run it on production.
+
+As an example, take the following Prisma schema with a `User` and `Post` model as a starting point:
+
+```prisma file=schema.prisma
+model Post {
+  id       Int     @id @default(autoincrement())
+  title    String  @db.VarChar(255)
+  content  String?
+  author   User    @relation(fields: [authorId], references: [id])
+  authorId Int
+}
+
+model User {
+  id    Int     @id @default(autoincrement())
+  name  String?
+  posts Post[]
+}
+```
+
+You will need to create the down migration first, before creating the corresponding up migration.
+
+### Generating the migrations
+
+1. Edit your Prisma schema to make the changes you require for your up migration. In this example, you will add a new `Profile` model:
+
+   ```prisma file=schema.prisma highlight=8-14;add|20;add
+   model Post {
+     id       Int     @id @default(autoincrement())
+     title    String  @db.VarChar(255)
+     content  String?
+     author   User    @relation(fields: [authorId], references: [id])
+     authorId Int
+   }
+   
+   model Profile {
+     id     Int     @id @default(autoincrement())
+     bio    String?
+     user   User    @relation(fields: [userId], references: [id])
+     userId Int     @unique
+   }
+   
+   model User {
+     id      Int      @id @default(autoincrement())
+     name    String?
+     posts   Post[]
+     profile Profile?
+   }
+   ```
+
+2. Generate the SQL file for the down migration. To do this, you will use `migrate diff` to make a comparison:
+
+   - from the newly edited schema
+   - to the state of the schema after the last migration
+
+   and output this to a SQL script.
+
+   There are two potential options for specifying the 'to' state:
+
+   - Using `--to-migrations`: this makes a comparison to the state of the migrations given in the migrations directory. This is the preferred option, as it is more robust, but it requires a shadow database. To use this option, run:
+
+     ```terminal
+     npx prisma migrate diff \
+      --from-schema-datamodel prisma/schema.prisma \
+      --to-migrations prisma/migrations --shadow-database-url $SHADOW_DATABASE_URL \
+      --script > down.sql
+     ```
+
+   - Using `--to-schema-datasource`: this makes a comparison to the state of the database. This does not require a shadow database, but it does rely on the database having an up-to-date schema. To use this option, run:
+
+     ```terminal
+     npx prisma migrate diff \
+      --from-schema-datamodel prisma/schema.prisma \
+      --to-schema-datasource prisma/schema.prisma \
+      --script > down.sql
+     ```
+
+3. Generate the up migration with a name of `add_profile`:
+
+   ```terminal
+   npx prisma migrate dev --name add_profile
+   ```
+
+   This will create a new `<timestamp>_add_profile` directory inside the `prisma/migrations` directory, with your new `migration.sql` up migration file inside.
```

Comment:
This will actually create and apply the migration

Reply:
I've changed to 'generate and apply', happy to include more detail if needed